src/index.ts

```diff
@@ -85,44 +82,44 @@ export function resolveBundle(
 
 				// Now that we have a resolver, add the new files needed to the stream
 				resolver
-					.resolve(bundle)
-					.then(additionalItems => {
-						return Promise.map(additionalItems, (file: FileInfo) => {
-							pack.entry({ name: file.name, size: file.size }, file.contents);
-							if (file.name === 'Dockerfile') {
-								return bundle.callDockerfileHook(file.contents.toString());
+				.resolve(bundle)
+				.then(additionalItems => {
+					return Promise.map(additionalItems, (file: FileInfo) => {
+						pack.entry({ name: file.name, size: file.size }, file.contents);
+						if (file.name === 'Dockerfile') {
+							return bundle.callDockerfileHook(file.contents.toString());
+						}
+					})
+					.then(() => {
+						return Promise.try(() => {
+							if (resolver.name === 'Standard Dockerfile') {
```

Comment:
This seems like quite a hack to rely on the resolver name, it means that for this particular special resolver we support calling dockerfile hooks for adding dockerfiles, but not for other resolves that add dockerfiles (eg the nodejs one) - in general it seems like we're missing certain support that allows resolvers to be fully isolated from the core
Beyond that it would be better to do `resolver instanceof DockerfileResolver` which should allow you to switch to just `resolver.getDockerfileContents()` below, without needing to force type casting